private/bufpkg/bufimage/util.go

```diff
@@ -433,9 +435,18 @@ func imageToCodeGeneratorRequest(
 			includeWellKnownTypes,
 		) {
 			request.FileToGenerate = append(request.FileToGenerate, imageFile.Path())
+			// Source-retention options for items in FileToGenerate are provided in SourceFileDescriptors.
+			request.SourceFileDescriptors = append(request.SourceFileDescriptors, fileDescriptorProto)
+			// And the corresponding descriptor in ProtoFile will have source-retention options stripped.
+			var err error
+			fileDescriptorProto, err = options.StripSourceRetentionOptionsFromFile(fileDescriptorProto)
+			if err != nil {
+				return nil, fmt.Errorf("failed to process %q for code generator request: %w", imageFile.Path(), err)
```

Comment:
`failed to strip source-retention options for file %q when constructing a CodeGeneratorRequest`

Reply:
Done